mod/likes/start.php

```diff
@@ -160,46 +137,86 @@ function likes_river_menu_setup($hook, $type, $return, $params) {
 		return;
 	}
 
-	$object = $item->getObjectEntity();
-	if (!$object || !$object->canAnnotate(0, 'likes')) {
+	$entity = $item->getObjectEntity();
+	if (!$entity) {
 		return;
 	}
 
-	$hasLiked = \Elgg\Likes\DataService::instance()->currentUserLikesEntity($object->guid);
+	$type = $entity->type;
+	$subtype = $entity->getSubtype();
+	$likable = (bool) elgg_trigger_plugin_hook('likes:is_likable', "$type:$subtype", [], false);
```

Comment:
in 2.3 this will come from the canAnnotate()... why not reuse that? so `$likable = $object->canAnnotate(0, 'likes')`